File: examples/availability/publish_ipfs/publish_ipfs.java
import com.bloock.sdk.Bloock;
import com.bloock.sdk.client.AvailabilityClient;
import com.bloock.sdk.client.RecordClient;
import com.bloock.sdk.entity.IpfsPublisher;
import com.bloock.sdk.entity.Record;

public class App {
  public static void main(String[] args) throws Exception {
    Bloock.apiKey = System.getenv("API_KEY");

    RecordClient recordClient = new RecordClient();
    AvailabilityClient availabilityClient = new AvailabilityClient();

    Record record = recordClient
        .fromString("Hello world")
        .build();

    String id = availabilityClient.publish(record, new IpfsPublisher());

    System.out.println("Record was published successfully");
  }
}

File: examples/availability/publish_ipfs/publish_ipfs.go
import (
	"fmt"
	"log"
	"os"

	"github.com/bloock/bloock-sdk-go/v2"
	"github.com/bloock/bloock-sdk-go/v2/client"
	"github.com/bloock/bloock-sdk-go/v2/entity"
)

func main() {
	bloock.ApiKey = os.Getenv("API_KEY")

	recordClient := client.NewRecordClient()
	availabilityClient := client.NewAvailabilityClient()

	record, err := recordClient.FromString("Hello world").Build()
	if err != nil {
		log.Fatalln(err)
	}

	id, err := availabilityClient.Publish(record, entity.NewIpfsPublisher())
	if err != nil {
		log.Fatalln(err)
	}

	fmt.Println("Record was published successfully")
}
